### PR TITLE
Refactor redis-client & parameterise config

### PR DESCRIPTION
**Pull request by chotai**

### CODE REVIEW

**Review by feedmypixel — APPROVED** [review]
LGTM 👍

File: src/server/common/helpers/redis-client.js [hunk 1]
```diff
@@ -1,63 +1,44 @@
...
 import { createLogger } from '~/src/server/common/helpers/logging/logger.js'
 
 /**
  * Setup Redis and provide a redis client
  *
  * Local development - 1 Redis instance
  * Environments - Elasticache / Redis Cluster with username and password
- *
+ * @param {RedisConfig} redisConfig - Redis config
  * @returns {Cluster | Redis}
```
Comment (colinrotherham) [inline src/server/common/helpers/redis-client.js:10]: Might be easier removing this temporarily until 

Currently the type `RedisConfig` isn't defined anywhere

```suggestion
```

File: src/server/common/helpers/redis-client.js [hunk 2]
```diff
@@ -1,63 +1,44 @@
 import IoRedis from 'ioredis'
```
Comment (colinrotherham) [inline src/server/common/helpers/redis-client.js:1]: We'll want to use the named imports `@returns` refers to by name

```suggestion
import { Cluster, Redis } from 'ioredis'
```

File: src/server/common/helpers/redis-client.js [hunk 3]
```diff
@@ -1,63 +1,44 @@
...
-      [
-        {
-          host,
-          port
-        }
-      ],
-      {
+  const client = redisConfig.useSingleInstanceCache
+    ? new IoRedis({ port, host, db, keyPrefix })
+    : new IoRedis.Cluster([{ host, port }], {
```
Comment (colinrotherham) [inline src/server/common/helpers/redis-client.js:21]: Using the named imports to match `@returns`

```suggestion
    ? new Redis({ port, host, db, keyPrefix })
    : new Cluster([{ host, port }], {
```

**Review by feedmypixel — APPROVED** [review]
LGTM 👍

**Pull request closed without merging**